File: Server/adapters/apihandler.ts
import axios from "axios";
import newsapi from "keys/newsapi.json";
import mapbox from "keys/mapbox.json";

async function nhlApiRequest(path: string) {
  const response = await axios.get(`http://statsapi.web.nhl.com${path}`);
  return response.data;
}

async function enhancedNhlApiRequest(path: string) {
  const response = await axios.get(`https://api.nhle.com${path}`);
  return response.data;
}

async function newsApiRequest(path: string) {
  const response = await axios.get(
    `http://newsapi.org${path}&apiKey=${newsapi.key}`
  );
  return response.data;
}

async function mapboxApiRequest(query: string): Promise<TApiMapboxLocations> {
  const response = await axios.get(
    `https://api.tiles.mapbox.com/geocoding/v5/mapbox.places/${encodeURIComponent(
      query
    )}.json?access_token=${mapbox.key}`
  );
  return response.data;
}

async function wikiApiRequest(query: string): Promise<TApiWikiResponse> {
  const response = await axios.get(
    `https://en.wikipedia.org/w/api.php?format=json&action=query&prop=extracts&exintro&explaintext&redirects=1&titles=${encodeURIComponent(
      query
    )}`
  );
  const pages = response.data.query.pages;
  const wikiResponse = pages[Object.keys(pages)[0]];

  return wikiResponse;
}

async function playerWikiRequest(query: string, optionalQuery = "") {
  let result = await wikiApiRequest(query);
  if (result.extract.includes("may refer to") && optionalQuery.length > 0) {
    result = await wikiApiRequest(`${query} (ice hockey)`);
    if (
      result.extract.includes("may refer to") ||
      Object.prototype.hasOwnProperty.call(result, "missing")
    ) {
      result = await wikiApiRequest(`${query} ${optionalQuery}`);
    }
  }
  return result;
}

async function wikiApiAdvancedRequest(mainQuery: string, subQuery: string) {
  let res = await wikiApiRequest(`${mainQuery} ${subQuery}`);
  if (res === undefined || res.ns == 0) {
    res = await wikiApiRequest(`${mainQuery}`);
  }
  return res;
}

export {
  nhlApiRequest,
  enhancedNhlApiRequest,
  newsApiRequest,
  mapboxApiRequest,
  wikiApiRequest,
  playerWikiRequest,
  wikiApiAdvancedRequest,
};

File: Database/apihandler.py
# importing the requests library
import requests
from urllib.parse import quote
import wikipediaapi
import json

wiki_wiki = wikipediaapi.Wikipedia('en')

keys_file = open("keys/mapbox.json")
keys = json.load(keys_file)

def nhl_api_request(path, params):
  urlPath = "http://statsapi.web.nhl.com{0}.".format(path)
  r = requests.get(url = urlPath, params = params)
  
  data = r.json()
  return data

def enhanced_nhl_api_request(path):
  urlPath = "http://api.nhle.com{0}.".format(path)
  r = requests.get(url = urlPath)
  
  data = r.json()
  return data

def mapbox_api_request(query):
  queryParam = quote(query.encode("utf-8"))
  url = "https://api.tiles.mapbox.com/geocoding/v5/mapbox.places/{0}.json".format(queryParam)
  r = requests.get(url = url, params = {
    "access_token": keys["key"]
  })
  
  data = r.json()
  return data

def wiki_api_request(query):
  queryParam = query.replace(" ", "_") #quote(query.encode("utf-8"))
  page_py = wiki_wiki.page(queryParam)
  return page_py

def wiki_api_advanced_request(mainQuery, subQuery):
  response = wiki_api_request("{0} {1}".format(mainQuery, subQuery))
  if(response.exists() == False):
    response = wiki_api_request(mainQuery)
  return response